### PR TITLE
Don't Require contiguous For Extern Kernels

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #88208
* __->__ #87650
* #87641

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @jansel @lezcano @fdrocha

### CODE REVIEW

File: torch/_inductor/ir.py [hunk 1]
```diff
@@ -2848,10 +2838,6 @@ def unflatten_args(new_tensor_args, new_non_tensor_args):
                     new_args.append(next(it_non_tensors))
             return pytree.tree_unflatten(new_args, args_spec)
 
-        tensor_args = [
-            cls.require_contiguous(cls.realize_input(x)) for x in tensor_args
```
Comment (jansel) [inline torch/_inductor/ir.py:2852]: Is `realize_input` still needed?

File: torch/_inductor/ir.py [hunk 2]
```diff
@@ -2848,10 +2838,6 @@ def unflatten_args(new_tensor_args, new_non_tensor_args):
...
-        ]
-
         # We don't have generic shape formulas, so just burn in the
         # shapes and run an example input.
         # TODO(jansel): replace this with dynamic shape formulas
```
Comment (ngimel) [inline torch/_inductor/ir.py:2843]: `tensor_args` is empty now, so how would `for x in tensor_args` work?

File: torch/_inductor/ir.py (hunk 1 above)
Comment (eellison) [reply torch/_inductor/ir.py:2852]: yea, good catch

**Review by desertfire — APPROVED** [review]

**Pull request closed without merging**